src/main/java/backend/Logic.java

```diff
@@ -36,8 +38,9 @@
     private static final String STRING_EMPTY = "";
     private static final char STRING_PERIOD = '.';
 
-    private static final String REGEX_NEITHER_ALPHANUMERIC_NOR_SPACE =
-        "[^ a-zA-Z0-9]+";
+    private static final String REGEX_AUTHOR_NAME_FORMAT = "(^[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])";
```

Comment:
Is it intended to fail if author string doesn't start with alphanumeric? 
Removing '^' from the regex would simply strip leading non-alphanumeric instead of failing.